### PR TITLE
fix(blob)!: tendermint compatible commitment proof json marshall

### PR DESCRIPTION
**Pull request by zvolin**
Fixes https://github.com/celestiaorg/celestia-node/issues/3918

**Please note: this PR is BREAKING** as commitment proof encoding has changed.

### CODE REVIEW

**Review by rach-id — COMMENTED** [review]
Thanks for opening this PR.

Do you think we will still need the JSON marshall/unmarshall after merging this https://github.com/celestiaorg/celestia-node/pull/3821 to main?

**Comment by zvolin** [comment]
yes, it uses `coretypes.RowProof`. Everything that comes from core (tendermint fork) should be serialized with tendermint json encoder / decoder as it treats 64 bit integers differently (stringifies them)

**Comment by rach-id** [comment]
thanks for your response @zvolin. And why is this change useful?

**Comment by rach-id** [comment]
After second thinking about this, I guess you're right, this needs fixing

**Comment by zvolin** [comment]
> And why is this change useful?

to have unified representation of objects regardless whether they are returned from celestia-node api or tendermint rest api. Moreover, tendermint implementations in different languages can have the rules how to encode types in different level than in go, namely [tendermint-rs bakes how the type is meant to be encoded on type level](https://github.com/informalsystems/tendermint-rs/blob/main/proto/src/prost/v0_34/tendermint.crypto.rs#L7), not encoder level, so in lumina we had to patch tendermint to achieve celestia-node compatible representation, but we could never be compatible with both tendermint and celestia-node.

tl;dr it originates from us wishing to drop tendermint-rs fork but I believe it's a good thing in general

**Comment by vgonkivs** [comment]
@zvolin could you please fix lint?

**Comment by zvolin** [comment]
switched to marshaling whole type as per https://github.com/celestiaorg/celestia-node/pull/3930#discussion_r1837870368

**Review by renaynay — APPROVED** [review]

**Comment by vgonkivs** [comment]
Linter failed.

**Comment by zvolin** [comment]
should be good now, sorry

**Review by renaynay — COMMENTED** [review]
@zvolin some broken tests

**Comment by vgonkivs** [comment]
` blob_fuzz_test.go:69: invalid 64-bit integer encoding "64", expected string`

**Review by vgonkivs — APPROVED** [review]
LGTM 🚀 . Thanks @zvolin

**Comment by zvolin** [comment]
`make test-integration SHORT=true TAGS=sync` passes for me locally :thinking:
